map.go

```diff
@@ -299,6 +321,16 @@ func (m *Map) lookup(key interface{}, valueOut internal.Pointer) error {
 	return errors.WithMessage(err, "lookup failed")
 }
 
+func (m *Map) lookupAndDelete(key interface{}, valueOut internal.Pointer) error {
```

Comment:
Is there a reason this code isn't just in `LookupAndDelete`?

Reply:
I wanted to follow the existing structure `lookup` does.
So, I've moved this code into `LookupAndDelete`